staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -697,6 +698,9 @@ func shouldSyncUpdatedNode(oldNode, newNode *v1.Node) bool {
 	if oldNode.Spec.ProviderID != newNode.Spec.ProviderID {
 		return true
 	}
+	if oldNode.UID != newNode.UID {
+		return true
+	}
```

Comment:
So we now have two places that check the same things.  What about something like:

```
type protoNode // outside any func

func distillNode(*v1.Node) protoNode {
	// ...
}
```

Then we can say here:

```
if distillNode(oldNode) != distillNode(newNode) {
```

and change `nodesSufficientlyEqual` like:

```
	if len(oldNodes) != len(newNodes) {
		return false
	}

	mOld := map[string]protoNode{}
	for _, n := range oldNodes {
		mOld[n.UID] = distillNode(n)
	}

	for _, n := range newNodes {
		if mOld[n.UID] != distillNode(n) {
			return false
		}
		delete(mOld, n.UID)
	}
	if len(mOld) != 0 {
		return false
	}

	return true
}
```